plugin/PDFDocumentPage.qml

```diff
@@ -108,30 +109,105 @@ DocumentPage {
             contextMenuLinks.url = linkTarget
             hook.showMenu(contextMenuLinks)
         }
-        clip: anchors.bottomMargin > 0
+        onAnnotationClicked: {
+            base.open = false
+            switch (annotation.type) {
+                case PDF.Annotation.Highlight:
+                contextMenuHighlight.annotation = annotation
+                hook.showMenu(contextMenuHighlight)
+                break
+                case PDF.Annotation.Caret:
+                case PDF.Annotation.Text:
+                pdfDocument.edit(annotation)
+                break
+                default:
+            }
+        }
+        onAnnotationLongPress: {
+            base.open = false
+            switch (annotation.type) {
+                case PDF.Annotation.Highlight:
+                contextMenuHighlight.annotation = annotation
+                hook.showMenu(contextMenuHighlight)
+                break
+                case PDF.Annotation.Caret:
+                case PDF.Annotation.Text:
+                contextMenuText.annotation = annotation
+                hook.showMenu(contextMenuText)
```

Comment:
Cases and actual code shouldn't start on same point. Qt convention would be moving case to match with switch.